#!/usr/bin/env python

PACKAGE = "aruco"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("trans_speed", double_t, 0, "Translational speed (m/s)",    0.15, 0,   1)
gen.add("rot_speed", double_t, 0, "Rotational speed (rad/s)",    1.0, 0,   2)
gen.add("trans_target", double_t, 0, "Distance (m) from marker the robot will maintain",  0.5, 0.3, 1)
gen.add("rot_target", double_t, 0, "Angular distance (rad) from marker the robot will maintain",  0.0, -2.0, 2.0)
gen.add("trans_tol", double_t, 0, "Translational tolerance (m)",  0.1, 0.0, 1)
gen.add("rot_tol", double_t, 0, "Rotational tolerance (rad)",  0.1, 0.0, 1)
gen.add("track_marker", int_t, 0, "Marker number to track", 406, 0, 1000)

exit(gen.generate(PACKAGE, "PBVS_controller", "PBVS_controller"))
